### PR TITLE
JS: Fix flow for nested destructurings

### PR DESCRIPTION
**Pull request by asger-semmle**
Evaluation pending...

### CODE REVIEW

File: javascript/ql/test/library-tests/TaintTracking/destruct.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+function test() {
+  
+  function f(obj) {
+    let { x: { y: { z } } } = obj;
+    sink(z); // NOT OK
```
Comment (xiemaisi) [inline javascript/ql/test/library-tests/TaintTracking/destruct.js]: Could we have another test to show/document that this flow only happens for taint tracking, not for normal data flow?

**Comment by asger-semmle** [comment]
[Evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/florian.ti.semmle.com_1558128129390) was completely uneventful.

File: javascript/ql/test/library-tests/TaintTracking/destruct.js (hunk 1 above)
Comment (asger-semmle) [reply javascript/ql/test/library-tests/TaintTracking/destruct.js]: Added a copy of this test that uses `DataFlow::Configuration`.

**Comment by asger-semmle** [comment]
Should be ready to land

**Pull request merged**